src/Cache/CacheOperationTypes/CacheOperationTypes.php

```diff
@@ -441,3 +475,209 @@ class CacheListPushBackResponseError extends CacheListPushBackResponse
     use ErrorBody;
 }
 
+abstract class CacheListPopFrontResponse extends ResponseBase
+{
+    public function asHit() : CacheListPopFrontResponseHit|null
+    {
+        if ($this->isHit())
+        {
+            return $this;
+        }
+        return null;
+    }
+
+    public function asMiss() : CacheListPopFrontResponseMiss|null
+    {
+        if ($this->isMiss())
+        {
+            return $this;
+        }
+        return null;
+    }
+
+    public function asError() : CacheListPopFrontResponseError|null
+    {
+        if ($this->isError())
+        {
+            return $this;
+        }
+        return null;
```

Comment:
I like these conveniences. For my benefit: did we align on returning `null` when calling an incompatible call vs throwing an exception?

Reply:
Yes, we did. This allows for pseudo-pattern matching and enforces the contract with the developer that once a request is made to the cache client all Momento errors will be returned and not thrown.